### PR TITLE
Computeenvironment  and jobqueue

### PR DESCRIPTION
**Pull request by sachinmalanki**
### Description of your changes

Creating new AWS Batch Resources - `ComputeEnvironment` and `JobQueue`
Continuing from - https://github.com/crossplane-contrib/provider-upjet-aws/pull/1489 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR if necessary.~

### How has this code been tested

This code has been tested locally and with uptest in the below [comments](https://github.com/crossplane-contrib/provider-upjet-aws/pull/1542#issuecomment-2442630973).

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by sachinmalanki** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by sachinmalanki** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by sachinmalanki** [comment]
The Uptest fails because of this error. 
`  logger.go:42: 13:57:17 | case/0-apply |     - lastTransitionTime: "2024-10-25T13:56:19Z"
    logger.go:42: 13:57:17 | case/0-apply |       message: 'update failed: async update failed: failed to set observation: v1beta1.ComputeEnvironmentObservation.ComputeResources:
    logger.go:42: 13:57:17 | case/0-apply |         readObjectStart: expect { or n, but found [, error found in #10 byte of ...|sources":[{"allocati|...,
    logger.go:42: 13:57:17 | case/0-apply |         bigger context ...|,"id":"sample","region":null,"compute_resources":[{"allocation_strategy":"","desired_vcpus":0,"max_v|...'
    logger.go:42: 13:57:17 | case/0-apply |       reason: ReconcileError
    logger.go:42: 13:57:17 | case/0-apply |       status: "False"
    logger.go:42: 13:57:17 | case/0-apply |       type: Synced
    logger.go:42: 13:57:17 | case/0-apply |     - lastTransitionTime: "2024-10-25T13:42:44Z"
    logger.go:42: 13:57:17 | case/0-apply |       reason: Available
    logger.go:42: 13:57:17 | case/0-apply |       status: "True"
    logger.go:42: 13:57:17 | case/0-apply |       type: Ready
    logger.go:42: 13:57:17 | case/0-apply |     - lastTransitionTime: "2024-10-25T13:56:19Z"
    logger.go:42: 13:57:17 | case/0-apply |       message: 'async update failed: failed to set observation: v1beta1.ComputeEnvironmentObservation.ComputeResources:
    logger.go:42: 13:57:17 | case/0-apply |         readObjectStart: expect { or n, but found [, error found in #10 byte of ...|sources":[{"allocati|...,
    logger.go:42: 13:57:17 | case/0-apply |         bigger context ...|,"id":"sample","region":null,"compute_resources":[{"allocation_strategy":"","desired_vcpus":0,"max_v|...'
    logger.go:42: 13:57:17 | case/0-apply |       reason: AsyncUpdateFailure
    logger.go:42: 13:57:17 | case/0-apply |       status: "False"
    logger.go:42: 13:57:17 | case/0-apply |       type: LastAsyncOperation`

Any idea on how to fix this? I can see that `compute resource` expects object but is provided an array which is likely the reason for this error.

File: config/batch/config.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+	p.AddResourceConfigurator("aws_batch_job_queue", func(r *config.Resource) {
+		r.References = config.References{
+			"compute_environment_order.compute_environment": config.Reference{
+				TerraformName: "aws_batch_compute_environment",
+				Extractor:     common.PathARNExtractor,
+			},
+		}
+	})
+}
```
Comment (turkenf) [inline config/batch/config.go:39]: I tested these two resources in my local, and except for the automatically defined references, defining the above seems sufficient. Could you please remove the above and add only the following configurations?
```
func Configure(p *config.Provider) {
	p.AddResourceConfigurator("aws_batch_compute_environment", func(r *config.Resource) {
		r.References["compute_resources.subnets"] = config.Reference{
			TerraformName: "aws_subnet",
		}
		r.References["compute_resources.security_group_ids"] = config.Reference{
			TerraformName: "aws_security_group",
		}
	})
}
```

File: examples/batch/v1beta1/jobqueue.yaml [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    computeEnvironmentOrder:
+      - computeEnvironmentSelector:
+          matchLabels:
+            testing.upbound.io/example-name: sample
+        order: 1
+    priority: 1
+    region: eu-west-1
+    state: ENABLED
+  providerConfigRef:
+    name: aws-provider-config-test
```
Comment (turkenf) [inline examples/batch/v1beta1/jobqueue.yaml:20]: We use the default provider config in uptest; could you please remove the above?

File: examples/batch/v1beta1/jobqueue.yaml [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+      subnetsRefs:
+      - name: sample
+      type: EC2
+    region: eu-west-1
+    serviceRoleSelector:
+      matchLabels:
+        testing.upbound.io/example-name: aws_batch_service_role
+    type: MANAGED
+  providerConfigRef:
+    name: aws-provider-config-test
```
Comment (turkenf) [inline examples/batch/v1beta1/jobqueue.yaml:55]: We use the default provider config in uptest; could you please remove the above?

**Comment by turkenf** [comment]
Could you please also remove the batch group from [externalnamenottested.go](https://github.com/crossplane-contrib/provider-upjet-aws/blob/7cd7083a87978d40583581b5e5d09b05236e2e1d/config/externalnamenottested.go#L33)

Thank you 🙏

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/iot/v1beta1/topicruledestination.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/computeenvironment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/jobqueue.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/batch/v1beta1/jobqueue.yaml"

File: examples/batch/v1beta1/jobqueue.yaml [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+  labels:
+    testing.upbound.io/example-name: sample
+  name: sample
+spec:
+  forProvider:
+    cidrBlock: 10.1.0.0/16
+    region: eu-west-1 
+
+
+
```
Comment (turkenf) [inline examples/batch/v1beta1/jobqueue.yaml:330]: Could you please remove extra lines?

File: examples/batch/v1beta1/testhooks/delete-computeenvironment.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+set -aeuo pipefail
+
+# SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+# Note(mbbush): AWS Batch  uses the IAM role assigned to the Compute Environment during deletion
+# of the Compute Environment.
+# Deletion of the Compute Environment takes several minutes, and if the role is deleted before
+# it finishes, then it doesn't have permission, so the VPC deletion fails
```
Comment (turkenf) [inline examples/batch/v1beta1/testhooks/delete-computeenvironment.sh:11]:
```suggestion
# it finishes, then it doesn't have permission, so the Compute Environment deletion fails
```

File: examples/batch/v1beta1/computeenvironment.yaml [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+              }
+          ]
+      }
+---
+apiVersion: iam.aws.upbound.io/v1beta1
+kind: RolePolicyAttachment
+metadata:
+  annotations:
+    meta.upbound.io/example-id: batch/v1beta1/computeenvironment
+    # uptest.upbound.io/pre-delete-hook: testhooks/delete-computeenvironment.sh
```
Comment (turkenf) [inline examples/batch/v1beta1/computeenvironment.yaml:102]: Could you please remove this and the following unused hooks completely?

**Review by turkenf — COMMENTED** [review]
Many thanks for your effort and quick actions @sachinmalanki. The tests were finally passed successfully 🎉 

Finally, a few cosmetic corrections would be nice before moving forward. I left a few comments for your consideration.

**Comment by sachinmalanki** [comment]
Thanks @turkenf  for your help. I have made the small changes suggested

**Review by jeanduplessis — APPROVED** [review]

**Pull request merged**